samcli/lib/build/workflow_config.py

```diff
@@ -222,40 +222,25 @@ def get_workflow_config(
         ) from ex
 
 
-def supports_build_in_container(config: CONFIG) -> Tuple[bool, Optional[str]]:
+def supports_specified_workflow(specified_workflow: str) -> bool:
     """
-    Given a workflow config, this method provides a boolean on whether the workflow can run within a container or not.
+    Given a specified workflow, returns whether it is supported in container builds,
+    can be used to overwrite runtime and get docker image or not
 
     Parameters
     ----------
-    config namedtuple(Capability)
-        Config specifying the particular build workflow
+    specified_workflow
+        Workflow specified in the template
 
     Returns
     -------
-    tuple(bool, str)
-        True, if this workflow can be built inside a container. False, along with a reason message if it cannot be.
+    bool
+        True, if this workflow is supported, can be used to overwrite runtime and get docker image
     """
 
-    def _key(c: CONFIG) -> str:
-        return str(c.language) + str(c.dependency_manager) + str(c.application_framework)
+    supported_specified_workflow = ["dotnet7"]
```

Comment:
Isn't this return `False` for every other runtime? In that case, it looks like we might be breaking their build workflow

Reply:
This wouldn't break others. The last parameter passed to `_build_function_on_container` and downstream is newly added. It allows builds such as `dotnet7` with `provided.al2` as runtime and a specified workflow to use the specified workflow to get docker image. This method returns `False` for others, so no specified workflow being passed down and they will use `runtime` to get docker image as previous.